sim/battle.js

```diff
@@ -2350,19 +2350,18 @@ class Battle extends Dex.ModdedDex {
 			this.add('-zbroken', target);
 		}
 
-		// Generations 5-7 all truncate damage to 16 bits as the last step of all
-		baseDamage = tr(baseDamage, 16);
-
+		// Generation 6-7 moves the check for minimum 1 damage after the final modifier...
 		if (this.gen !== 5 && !baseDamage) return 1;
 
-		return baseDamage;
+		// ...but 16-bit truncation happens even later, and can truncate to 0
+		return tr(baseDamage, 16);
 	}
 
 	/**
 	 * @param {number} baseDamage
 	 */
 	randomizer(baseDamage) {
-		return this.trunc(baseDamage * (100 - this.random(16)) / 100);
+		return this.trunc(this.trunc(baseDamage * (100 - this.random(16))) / 100);
```

Comment:
```suggestion
		const tr = this.trunc;
		return tr(tr(baseDamage * (100 - this.random(16))) / 100);
```

With two calls to `this.trunc`, that's enough for me to want to abbreviate it.